waku/waku_filter_v2/subscriptions.nim

```diff
@@ -43,3 +43,11 @@ proc removePeers*(subscriptions: var FilterSubscriptions, peerIds: seq[PeerID])
   ## Remove all subscriptions for a given list of peers
   for peerId in peerIds:
     subscriptions.removePeer(peerId)
+
+proc containsAny*(criteria: FilterCriteria, otherCriteria: FilterCriteria): bool =
+  ## Check if a given pubsub topic is contained in a set of filter criteria
+  ## TODO: Better way to achieve this?
+  for criterion in otherCriteria:
+    if criterion in criteria:
+      return true
+  false
```

Comment:
Maybe we could make something like the following:
```suggestion
  let interesctionResult = criteria * otherCriteria
  return interesctionResult.len > 0
```

I don't know which option is faster. If possible, we could benchmark that in a separate tiny Nim example.

Reply:
Just ran a test, and I get the for-if solution is 5~ times faster than the intersection one. I guess we're going with the first 😆